devp2p/peer.py

```diff
@@ -101,17 +102,22 @@ def has_protocol(self, protocol):
 
     def receive_hello(self, proto, version, client_version_string, capabilities,
                       listen_port, nodeid):
-        # register in common protocols
         log.info('received hello', version=version,
                  client_version=client_version_string, capabilities=capabilities)
         self.remote_client_version = client_version_string
 
+        # enable backwards compatibility for legacy peers
+        if version < 5:
+            self.offset_based_dispatch = True
+            max_window_size = 2**32 # disable chunked transfers
+
```

Comment:
This is temporary, a better approach might be to enable legacy mode based on the encryption handshake changes (`rlpx_version`).